From cfef9413f7ec2691921e70b3000b5e75f263f1fb Mon Sep 17 00:00:00 2001
From: Vitaly Margolin <vmargolin@habana.ai>
Date: Tue, 18 Feb 2025 11:39:17 +0200
Subject: [PATCH] [SW-219450] g3: create itbs to set product in eeprom

Create files gaudi3-eeprom-set-product-hl325.itb and
gaudi3-eeprom-set-product-hl328.itb  that will write
the appropriate System Product HL value to eeprom

Change-Id: I9b64cd26fd720c686425b8754f992402423fc2b3
Signed-off-by: Vitaly Margolin <vmargolin@habana.ai>
---

diff --git a/scripts/make/gaudi3/Makefile b/scripts/make/gaudi3/Makefile
index 93554da..bf45235 100644
--- a/scripts/make/gaudi3/Makefile
+++ b/scripts/make/gaudi3/Makefile
@@ -86,6 +86,7 @@
 	@echo "default eeprom created"
 
 EEPROM_DATA_FILE = $(IMAGES_DIR)/eeprom-recovery.txt
+EEPROM_SET_PRODUCT = eeprom-set-product
 .PHONY: gaudi3-eeprom-set-recovery
 gaudi3-eeprom-set-recovery:
 	@echo "Layout Version=6" > $(EEPROM_DATA_FILE)
@@ -96,11 +97,28 @@
 	@echo "Scratch Pad=0000" >> $(EEPROM_DATA_FILE)
 	@echo "Board Description=Warning Recovery values Update EEPROM" >> $(EEPROM_DATA_FILE)
 
+.PHONY: gaudi3-eeprom-set-product-325
+gaudi3-eeprom-set-product-325:
+	@echo "System Product HL=145" > $(IMAGES_DIR)/$(EEPROM_SET_PRODUCT).txt
+
+.PHONY: gaudi3-eeprom-set-product-328
+gaudi3-eeprom-set-product-328:
+	@echo "System Product HL=148" > $(IMAGES_DIR)/$(EEPROM_SET_PRODUCT).txt
+
 RECOVERY_PROP_NAME = $(shell grep -Eo 'HL_FIT_RECOVERY_PROP[[:space:]]+\"[[:alnum:]]+' $(SPECS_EMBEDDED_DIR)/hl_fit_if.h | cut -d\" -f2)
 .PHONY: gaudi3-eeprom-create-recovery
 gaudi3-eeprom-create-recovery: gaudi3-eeprom-set-recovery
 	$(HOST_MAKE_ENV) ${BR2_EXTERNAL}/board/habanalabs/common/create-eeprom-fit.sh 1 $(RECOVERY_PROP_NAME) recovery
 	@echo "recovery eeprom created"
+
+.PHONY: gaudi3-eeprom-create-product-325
+gaudi3-eeprom-create-product-325 : gaudi3-eeprom-set-product-325
+	$(HOST_MAKE_ENV) ${BR2_EXTERNAL}/board/habanalabs/common/create-eeprom-fit.sh 1 $(RECOVERY_PROP_NAME) set-product
+	mv ${ITBS_DIR}/${DEV_PREFIX}-$(EEPROM_SET_PRODUCT).itb ${ITBS_DIR}/${DEV_PREFIX}-$(EEPROM_SET_PRODUCT)-hl325.itb
+.PHONY: gaudi3-eeprom-create-product-328
+gaudi3-eeprom-create-product-328 : gaudi3-eeprom-set-product-328
+	$(HOST_MAKE_ENV) ${BR2_EXTERNAL}/board/habanalabs/common/create-eeprom-fit.sh 1 $(RECOVERY_PROP_NAME) set-product
+	mv ${ITBS_DIR}/${DEV_PREFIX}-$(EEPROM_SET_PRODUCT).itb ${ITBS_DIR}/${DEV_PREFIX}-$(EEPROM_SET_PRODUCT)-hl328.itb
 
 EFUSE_DATA_FILE = $(IMAGES_DIR)/efuse_spare_bit_write.txt
 .PHONY: gaudi3-efuse-write-set-default
@@ -129,8 +147,9 @@
 	${DEV_PREFIX}-slt_set.itb			\
 	${DEV_PREFIX}-fake_fuse.itb			\
 	${DEV_PREFIX}-rr_dump.itb                       \
-	${DEV_PREFIX}-copy_efuse_to_fake_efuse.itb
-
+	${DEV_PREFIX}-copy_efuse_to_fake_efuse.itb	\
+	${DEV_PREFIX}-eeprom-set-product-hl325.itb	\
+	${DEV_PREFIX}-eeprom-set-product-hl328.itb
 
 .PHONY: gaudi3-internal-fit-create
 gaudi3-internal-fit-create: gaudi3-eeprom-create-default	\
@@ -147,7 +166,9 @@
 				gaudi3-slt-set-create				\
 				gaudi3-fake-fuse-create				\
 				gaudi3-rr-dump-create                           \
-				gaudi3-copy-efuse-to-fake-efuse-create
+				gaudi3-copy-efuse-to-fake-efuse-create		\
+				gaudi3-eeprom-create-product-325		\
+				gaudi3-eeprom-create-product-328
 
 	@$(call hl-check-files,$(GAUDI3_INTERNAL_FITS))
 	@tar czf $(IMAGES_DIR)/${DEV_PREFIX}-int-fit.tgz -C $(ITBS_DIR)/ $(GAUDI3_INTERNAL_FITS)
